src/ts/InternalBitmovinYospacePlayer.ts

```diff
@@ -903,15 +906,15 @@ export class InternalBitmovinYospacePlayer implements BitmovinYospacePlayerAPI {
       return DEFAULT_AD_BREAK_TEMPLATE_OBJECT;
     }
     return {
-      id: ysAdBreak.getIdentifier(), // can be null
+      id: ysAdBreak.getIdentifier() === null ? '' : ysAdBreak.getIdentifier()!, // WARNING: is this logic correct? i.e returning "" id if null
```

Comment:
I just noticed that the Bitmovin Player API defines that the ad break id is generated if none is available (in the context of VMAP client-side ads, specifically). To stay close to the API definition, I'd suggest something like this:

Add
```ts
  private generateAdBreakId(ysAdBreak: AdBreak): string {
    const ysAdBreakId = ysAdBreak.getIdentifier();
    if (ysAdBreakId) {
      return ysAdBreakId;
    }

    // The Bitmovin Player API defines that the ad break id is generated if non is available, doing the same for
    // Yospace ads.
    return `${ysAdBreak.getPosition()}-${toSeconds(ysAdBreak.getStart())}-${toSeconds(ysAdBreak.getDuration())}`;
  }
```
and use it here:
```suggestion
      id: this.generateAdBreakId(ysAdBreak),
```

Reply:
thank you; updated w/ the suggestions.